tck/src/main/java/org/reactivestreams/tck/TestEnvironment.java

```diff
@@ -1025,22 +1072,44 @@ public void expectCompletion(long timeoutMillis, String errorMsg) throws Interru
       } // else, ok
     }
 
-    @SuppressWarnings("unchecked")
+    /**
+     * @deprecated Deprecated in favor of {@link #expectError(Class, long, long, String)}.
+     */
+    @Deprecated
     public <E extends Throwable> E expectError(Class<E> clazz, long timeoutMillis, String errorMsg) throws Exception {
-      Thread.sleep(timeoutMillis);
-
-      if (env.asyncErrors.isEmpty()) {
-        return env.flopAndFail(String.format("%s within %d ms", errorMsg, timeoutMillis));
-      } else {
-        // ok, there was an expected error
-        Throwable thrown = env.asyncErrors.remove(0);
+      return expectError(clazz, timeoutMillis, timeoutMillis, errorMsg);
+    }
 
-        if (clazz.isInstance(thrown)) {
-          return (E) thrown;
+    @SuppressWarnings("unchecked")
+    final <E extends Throwable> E expectError(Class<E> clazz, final long totalTimeoutMillis,
+                                              long pollTimeoutMillis,
```

Comment:
nitpick: consistency on the `final` in the params; since its `long`s let's keep the signature similar-er as it was: no final on the `final long totalTimeoutMillis` as it cannot be modified anyway

Reply:
The `final`ness isn't meant to highlight that the caller's value won't be modified (primitives are passed by value as you eluded to), but to make sure this method preserves the value because it is used in the exception when a timeout occurs and the original value is expected. The `final` is meant to highlight the original value should be preserved as opposed to using it for intermediate calculations (e.g. instead of the `totalTimeoutRemainingMillis`  temp copy). I can remove it if you prefer because functionally it would be equivalent with the current code.

Reply:
I see, thanks for the explanation, let's keep then 👍